build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/types/DockerImage.java

```diff
@@ -146,35 +143,40 @@ private void setDependenciesAndArtifacts(Module buildInfoModule, Properties prop
             if (layer == null) {
                 continue;
             }
-            HttpResponse httpResponse = propertyChangeClient.executeUpdateFileProperty(layer.getFullPath(), artifactsPropsStr);
-            validateResponse(httpResponse);
-            Artifact artifact = new ArtifactBuilder(layer.getFileName()).sha1(layer.getSha1()).addProperties(properties).build();
+            Artifact artifact = new ArtifactBuilder(layer.getFileName()).sha1(layer.getSha1()).build();
             artifacts.add(artifact);
         }
         buildInfoModule.setArtifacts(new ArrayList<>(artifacts));
     }
 
+    private void setDependencies(Module buildInfoModule) throws IOException {
+        LinkedHashSet<Dependency> dependencies = new LinkedHashSet<>();
+        // Filter out duplicate layers from manifest by using HashSet.
+        // Docker manifest may hold 'empty layers', as a result, docker promote will fail to promote the same layer more than once.
+        for (String digest : DockerUtils.getLayersDigests(manifest)) {
+            DockerLayer layer = layers.getByDigest(digest);
+            Dependency dependency = new DependencyBuilder().id(layer.getFileName()).sha1(layer.getSha1()).build();
+            dependencies.add(dependency);
+        }
+        buildInfoModule.setDependencies(new ArrayList<>(dependencies));
+    }
+
     /**
      * Prepare AQL query to get all the manifest layers from Artifactory.
      * Needed for build-info sha1/md5 checksum for each artifact and dependency.
-     *
-     * @return
-     * @throws IOException
      */
-    private String getAqlQuery(ArtifactoryBuildInfoClient propertyChangeClient) throws IOException {
-        boolean includeVirtualRepos = propertyChangeClient.getArtifactoryVersion().isAtLeast(VIRTUAL_REPOS_SUPPORTED_VERSION);
+    private String getAqlQuery(boolean includeVirtualRepos, String Repo) throws IOException {
         List<String> layersDigest = DockerUtils.getLayersDigests(manifest);
         StringBuilder aqlRequestForDockerSha = new StringBuilder("items.find({")
-                .append("\"path\":\"").append(imagePath).append("\",\"$or\":[");
-        List<String> layersQuery = new ArrayList<String>();
+                .append("\"repo\":\"").append(Repo).append("\",\"$or\":[");
+        List<String> layersQuery = new ArrayList<>();
         for (String digest : layersDigest) {
             String shaVersion = DockerUtils.getShaVersion(digest);
             String singleFileQuery;
             if (StringUtils.equalsIgnoreCase(shaVersion, "sha1")) {
-                ;
                 singleFileQuery = String.format("{\"actual_sha1\": \"%s\"}", DockerUtils.getShaValue(digest));
```

Comment:
AQL strings should be as short as possible
```suggestion
                singleFileQuery = String.format("{\"actual_sha1\":\"%s\"}", DockerUtils.getShaValue(digest));
```